deps/thpool/thpool.c

```diff
@@ -94,11 +94,12 @@ typedef struct redisearch_thpool_t {
   pthread_mutex_t thcount_lock;     /* used for thread count etc */
   pthread_cond_t threads_all_idle;  /* signal to thpool_wait     */
   priority_queue jobqueue;          /* job queue                 */
+  LogFunc log;                      /* log callback              */
 } redisearch_thpool_t;
 
 /* ========================== PROTOTYPES ============================ */
 
-static int thread_init(redisearch_thpool_t* thpool_p, struct thread** thread_p, int id, LogFunc log);
+static int thread_init(redisearch_thpool_t* thpool_p, struct thread** thread_p, int id);
```

Comment:
nitpick - Move `log` to `redisearch_thpool_create` (and log error using it instead of `err()`)